solidity/contracts/core/AutomationVault.sol

```diff
@@ -182,13 +189,34 @@ contract AutomationVault is IAutomationVault {
           emit ApproveJob(_jobData.job);
         }
 
-        // Get the length of the selectors array
-        _selectorsLength = _jobData.functionSelectors.length;
+        // Get the array of selectors data
+        SelectorData[] memory _selectorsData = _jobData.selectorsData;
+
+        // Get the length of the selectors data array
+        _selectorsLength = _selectorsData.length;
 
         // Set the selectors for the job
         for (_j = 0; _j < _selectorsLength;) {
-          if (_approvedJobSelectors[_relay][_jobData.job].add(_jobData.functionSelectors[_j])) {
-            emit ApproveJobSelector(_jobData.job, _jobData.functionSelectors[_j]);
+          // Check if selector already exists if it does, skip it
+          if (
+            _approvedJobSelectorsWithHooks[_relay][_jobData.job][_selectorsData[_j].selector].selectorType
+              == JobSelectorType.DISABLED
+          ) {
+            // Add the selector to the list of selectors
+            _approvedJobSelectorsList[_relay][_jobData.job].push(_selectorsData[_j].selector);
+
+            // Add the selector to the list of selectors with hooks
+            _approvedJobSelectorsWithHooks[_relay][_jobData.job][_selectorsData[_j].selector] =
+              _selectorsData[_j].hookData;
```

Comment:
we should check that the inputted selectorsData isn't DISABLED? or we allow this?